pkg/capi/components/CCVariables/index.vue

```diff
@@ -0,0 +1,249 @@
+
+        definitions.forEach((definition: any) => {
+          const matchMachines = definition?.selector?.matchResources?.[matchKey]?.names || [];
+
+          if (matchMachines.includes(matchName)) {
+            out.push(...definition.jsonPatches);
+          }
+        });
+      });
+
+      return out;
+    },
+  },
+
+  methods: {
+    valueFor(variableDef: ClusterClassVariable) {
+      return (this.value.find((variable: any) => variable.name === variableDef.name) || {}).value;
+    },
+
+    updateVariables(val: any, variableDef: ClusterClassVariable) {
+      const out = [...this.value] as Array<any>;
+      const existingIdx = this.value.findIndex((variable: any) => variable.name === variableDef.name);
+
+      if (existingIdx >= 0) {
+        out[existingIdx].value = val;
+      } else {
+        out.push({ value: val, name: variableDef.name });
+      }
+      this.$emit('input', out);
+    },
+
+    /**
+   * When the cluster class changes, update the variables array:
+   *  remove any variables not defined in the new cluster class
+   *  if a variable is defined in both cluster classes, clear out the old default
+   *  set default values from the new cluster class definitions
+   *  if a variable is defined in both old and new cluster classes, and the user has set its value to something other than the old default, preserve that
+   */
+    updateVariableDefaults(neu: ClusterClassVariable[], old: ClusterClassVariable[]) {
+      const out = [...this.value].reduce((acc: CapiClusterVariable[], existingVar: CapiClusterVariable) => {
+        const neuDef = (neu || []).find(n => n.name === existingVar.name);
+
+        if (!neuDef) {
+          return acc;
+        }
+        const oldDefault = (old || []).find(d => d.name === existingVar.name)?.schema?.openAPIV3Schema?.default;
+
+        if (isDefined(oldDefault) && existingVar.value === oldDefault) {
+          delete existingVar.value;
+        }
+        let newDefault = neuDef.schema?.openAPIV3Schema?.default;
+
+        if (neuDef.schema?.openAPIV3Schema?.type === 'boolean' && !newDefault) {
+          newDefault = false;
+        }
+        if (isDefined(newDefault) && !existingVar.value) {
+          existingVar.value = newDefault;
+        }
+        acc.push(existingVar);
+
+        return acc;
+      }, []);
+
+      neu.forEach((def) => {
+        let newDefault = def.schema?.openAPIV3Schema?.default;
+
+        if (def.schema?.openAPIV3Schema?.type === 'boolean' && !newDefault) {
+          newDefault = false;
+        }
+        if (isDefined(newDefault) && !out.find(v => v.name === def.name)) {
+          out.push({ name: def.name, value: newDefault });
+        }
+      });
+
+      this.errorCount = 0;
+      this.$emit('input', out);
+    },
+
+    updateErrors(isValid: boolean) {
+      if (!isValid) {
+        this.errorCount++;
+      } else {
+        this.errorCount--;
+      }
+    },
+
+    newComponentType(variableDef: ClusterClassVariable, i: number) {
```

Comment:
I'm not crazy about this nor the re-render key hack, but I found it absurdly convoluted to create a new row for each new component type using purely scss; this is easy to follow ultimately.